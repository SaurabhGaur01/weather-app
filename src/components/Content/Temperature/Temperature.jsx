import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import Typography from '@material-ui/core/Typography';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import { temperatureClass } from '../../../utils/getStyles';
import handleRetrieveWeatherInfo from '../../../thunks/handleRetrieveWeatherInfo';
import { TEMPERATURE_IN_FAHRENHEIT, DEFAULT_LOCATION, TEMPERATURE_IN_CELSIUS } from '../../../constants/apiInfo';

const Temperature = ({ actionCallWeatherApi }) => {
  const classes = temperatureClass();
  const [value, setValue] = React.useState('F');

  const handleChange = (event) => {
    setValue(event.target.value);
    if(event.target.value === 'C'){
      actionCallWeatherApi(TEMPERATURE_IN_CELSIUS, DEFAULT_LOCATION);
    }else{
      actionCallWeatherApi(TEMPERATURE_IN_FAHRENHEIT, DEFAULT_LOCATION);
    }
  };

  return (
    <React.Fragment>
      <Typography variant="h6">
        {DEFAULT_LOCATION}
      </Typography>
      <hr/>
      <FormControl component="fieldset" className={classes.formControl}>
        <RadioGroup row aria-label="position" name="position" value={value} onChange={handleChange}>
          <FormControlLabel
            value="C"
            control={<Radio color="primary" />}
            label="Celcius"
            labelPlacement="end"
            className={classes.labelControl}
          />
          <FormControlLabel 
              value="F" 
              control={<Radio color="primary" />} 
              label="Fahrenheit" 
              labelPlacement="end"
          />
        </RadioGroup>
      </FormControl>
    </React.Fragment>
  );
}

Temperature.propTypes = {
  actionCallWeatherApi: PropTypes.func.isRequired,
};

const mapDispatchAsProps = dispatch =>{
  return {
      actionCallWeatherApi: (temperature, location) => dispatch(handleRetrieveWeatherInfo(temperature, location)),
  };
}

const hocChain = compose(
  connect(null, mapDispatchAsProps),
);

export { Temperature as TestableTemperature};

export default hocChain(Temperature);