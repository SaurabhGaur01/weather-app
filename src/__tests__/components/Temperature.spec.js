import React from 'react';
import { shallow } from 'enzyme';
import { TestableTemperature } from '../../components/Content/Temperature/Temperature';

jest.mock('@material-ui/core/Typography', () => 'span');
jest.mock('@material-ui/core/Radio', () => 'radio')
jest.mock('@material-ui/core/RadioGroup', () => 'RadioGroup')
jest.mock('@material-ui/core/FormControlLabel', () => 'label')
jest.mock('@material-ui/core/FormControl', () => 'form')

describe('Temperature Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            actionCallWeatherApi: jest.fn(),
            handleChange: () => ('handleChange'),
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableTemperature {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });
})