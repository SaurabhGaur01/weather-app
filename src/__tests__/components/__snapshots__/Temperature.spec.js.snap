// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature Component should render correctly 1`] = `
<React.Fragment>
  <span
    variant="h6"
  >
    Munich,de
  </span>
  <hr />
  <form
    component="fieldset"
  >
    <RadioGroup
      aria-label="position"
      name="position"
      onChange={[Function]}
      row={true}
      value="F"
    >
      <label
        className="makeStyles-labelControl-1"
        control={
          <radio
            color="primary"
          />
        }
        label="Celcius"
        labelPlacement="end"
        value="C"
      />
      <label
        control={
          <radio
            color="primary"
          />
        }
        label="Fahrenheit"
        labelPlacement="end"
        value="F"
      />
    </RadioGroup>
  </form>
</React.Fragment>
`;
